org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaThread.scala

```diff
@@ -330,8 +334,11 @@ private[model] class ScalaThreadActor private(thread: ScalaThread) extends BaseD
       thread.threadRef.resume()
     case DropToFrame(frame) =>
       thread.dropToFrameInternal(frame)
-    case RebindStackFrames =>
-      if (thread.isSuspended) thread.rebindScalaStackFrames()
+    case RebindStackFrames(shouldFireChangeEvent) =>
+      if (thread.isSuspended) {
+        thread.rebindScalaStackFrames()
+        if (shouldFireChangeEvent) thread.fireChangeEvent(DebugEvent.CONTENT)
```

Comment:
Is this enough to refresh all the other windows? If whoever triggers this event causes some stack frames to be dropped (I don't know if that's the case), then also the Threads view and the Source editor need to be refreshed.